### ISSUE TITLE
Build with vmware-iso reports unexpected EOF error

### ISSUE THREAD

**Issue by MAPerk** [hashicorp/packer#5915]
FOR BUGS:

Describe the problem and include the following information:

Packer Version is 1.2.0
Host platform is VMWARE 6.5.0

Below is a cut of the debug log (PACKER_LOG=1 packer build)

<<<=============>>>
....
```
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 Serving a plugin connection...
2018/02/16 07:24:36 ui: ESC[1;32mubuntu-1604-base output will be in this color.ESC[0m
2018/02/16 07:24:36 ui: 
2018/02/16 07:24:36 Build debug mode: false
2018/02/16 07:24:36 Force build: false
2018/02/16 07:24:36 On error: 
2018/02/16 07:24:36 Preparing build: ubuntu-1604-base
2018/02/16 07:24:36 Waiting on builds to complete...
2018/02/16 07:24:36 Starting build run: ubuntu-1604-base
2018/02/16 07:24:36 Running builder: vmware-iso
2018/02/16 07:24:36 [INFO] (telemetry) Starting builder vmware-iso
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 reconnecting to TCP connection for SSH
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 handshaking with SSH
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 Keyboard interactive challenge:
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 -- User:
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 -- Instructions:
2018/02/16 07:24:36 packer: 2018/02/16 07:24:36 -- Question 1: Password:
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 Keyboard interactive challenge:
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 -- User:
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 -- Instructions:
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 handshake complete!
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 opening new ssh session
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 [INFO] agent forwarding enabled
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 opening new ssh session
2018/02/16 07:24:37 packer: 2018/02/16 07:24:37 starting remote command: esxcli --formatter csv system version get
2018/02/16 07:24:37 packer: panic: non-positive interval for NewTicker
2018/02/16 07:24:37 packer: 
2018/02/16 07:24:37 packer: goroutine 168 [running]:
2018/02/16 07:24:37 packer: time.NewTicker(0x0, 0x0)
2018/02/16 07:24:37 packer:     /usr/local/go/src/time/tick.go:23 +0x1d0
2018/02/16 07:24:37 packer: github.com/hashicorp/packer/communicator/ssh.(*comm).Start.func1(0xc4201f5590, 0xc42020b950)
2018/02/16 07:24:37 packer:     /Users/mwhooker/go/src/github.com/hashicorp/packer/communicator/ssh/communicator.go:118 +0x46
2018/02/16 07:24:37 packer: created by github.com/hashicorp/packer/communicator/ssh.(*comm).Start
2018/02/16 07:24:37 packer:     /Users/mwhooker/go/src/github.com/hashicorp/packer/communicator/ssh/communicator.go:114 +0x209
2018/02/16 07:24:37 [INFO] (telemetry) ending vmware-iso
2018/02/16 07:24:37 [INFO] (telemetry) found error: unexpected EOF
2018/02/16 07:24:37 ui error: ESC[1;31mBuild 'ubuntu-1604-base' errored: unexpected EOFESC[0m
2018/02/16 07:24:37 Builds completed. Waiting on interrupt barrier...
2018/02/16 07:24:37 machine readable: error-count []string{"1"}
2018/02/16 07:24:37 ui error: 
==> Some builds didn't complete successfully and had errors:
2018/02/16 07:24:37 machine readable: ubuntu-1604-base,error []string{"unexpected EOF"}
2018/02/16 07:24:37 ui error: --> ubuntu-1604-base: unexpected EOF
```
<<<=============>>>
will post the full log to the gist (once I figure that out)
EDIT: link in GIST.
https://gist.github.com/MAPerk/c5a30994044d85b0e4257b5cdea65d9e#file-crash-log

Here is my JSON build file.
<<<=============>>>
```json
{
  "builders": [
    {
      "boot_command": [
        "<esc><wait>",
        "install ",
        "preseed/url=http://nullgrid.s3.amazonaws.com/debian.cfg ",
        "debian-installer=en_US auto=true locale=en_US kbd-chooser/method=us ",
        "netcfg/get_hostname={{ .Name }} ",
        "netcfg/get_domain=nullgrid.net ",
        "fb=false debconf/frontend=noninteractive ",
        "console-setup/ask_detect=false console-keymaps-at/keymap=us ",
        "keyboard-configuration/xkb-keymap=us ",
        "<enter>"
      ],
      "disk_type_id": "thin",
      "guest_os_type": "debian8-64",
      "headless": true,
      "iso_checksum": "9479c5c2df72ae3878116c43fb42eefae53d1fe363ce514a6afc8289064b9f5f",
      "iso_checksum_type": "sha256",
      "iso_url": "http://cdimage.debian.org/debian-cd/8.6.0/amd64/iso-cd/debian-8.6.0-amd64-netinst.iso",
      "keep_registered": true,
      "name": "debian-8-base",
      "output_directory": "debian-8-base",
      "remote_datastore": "{{user `esxi_datastore`}}",
      "remote_host": "{{user `esxi_host`}}",
      "remote_password": "{{user `esxi_password`}}",
      "remote_type": "esx5",
      "remote_username": "{{user `esxi_username`}}",
      "shutdown_command": "echo 'shutdown -h now' > shutdown.sh; echo 'nullgrid'|sudo -S sh 'shutdown.sh'",
      "ssh_password": "nullgrid",
      "ssh_timeout": "15m",
      "ssh_username": "nullgrid",
      "tools_upload_flavor": "linux",
      "type": "vmware-iso",
      "vm_name": "debian-8-base",
      "vmx_data": {
        "ethernet0.networkName": "VM Network INT"
      }
    }
  ],
  "post-processors": [],
  "provisioners": [
    {
      "execute_command": "echo 'nullgrid' | {{ .Vars }} sudo -E -S bash '{{ .Path }}'",
      "scripts": [
        "scripts/open-vm-tools.sh"
      ],
      "type": "shell"
    }
  ]
}
```
<<<=============>>>

thanks,
Mike P.

**Comment by mporrato** [comment]
Had the same error here: commit a4123eabe35e3d29e3aa6d50efa107800aa87cb6 solved the problem, but it crashes with a SIGSEV further on.

**Comment by mporrato** [comment]
@SwampDragons not yet, I'm trying to figure out if it's a PEBKAC before raising the ticket

**Comment by SwampDragons** [comment]
Sounds good -- let us know here either way.

**Comment by mporrato** [comment]
@SwampDragons Ok, can't see any obvious mistake in my process, so here it is: #5916

**Comment by mporrato** [comment]
@MAPerk : the patch for #5916 seems to fix the broken ESXi support in 1.2.0: you should be able to use your template with that.

**Comment by mwhooker** [comment]
Gonna close this since it's fixed in master. Thanks for the issue!
